src/lib.rs

```diff
@@ -572,67 +477,198 @@ impl KafkaJsonToDelta {
         should
     }
 
-    async fn init_offsets(
-        &mut self,
-        offsets: &HashMap<DataTypePartition, DataTypeOffset>,
+    async fn consume_value_buffers(
+        &self,
+        state: &mut ProcessingState,
+    ) -> Result<(Vec<Value>, HashMap<DataTypePartition, DataTypeOffset>), KafkaJsonToDeltaError>
+    {
+        let mut partition_assignment = self.partition_assignment.lock().await;
+
+        let ConsumedBuffers {
+            values,
+            partition_offsets,
+        } = state.value_buffers.consume();
+
+        partition_assignment.update_offsets(&partition_offsets);
+
+        let partition_offsets = partition_assignment.partition_offsets();
+
+        Ok((values, partition_offsets))
+    }
+
+    async fn build_actions(
```

Comment:
does this need to be async? I'm not seeing an await.

Reply:
No it's not, just a leftover from the copypasting